salt/minion.py

```diff
@@ -1594,7 +1594,8 @@ def _thread_return(cls, minion_instance, opts, data):
                     # this minion is blacked out. Only allow saltutil.refresh_pillar and the whitelist
                     if function_name != 'saltutil.refresh_pillar' and function_name not in whitelist:
                         minion_blackout_violation = True
-                elif minion_instance.opts['grains'].get('minion_blackout', False):
+                # minion_blackout_whitelist by grains has higher priority than by pillar
```

Comment:
I disagree with this comment. By the code now there is no any priority, if blackout mode is enabled and function isn't whitelisted it's blocked.